packages/next-swc/crates/next-custom-transforms/src/transforms/page_static_info/collect_exported_const_visitor.rs

```diff
@@ -0,0 +1,210 @@
+    }
+}
+
+impl VisitAstPath for CollectExportedConstVisitor {
+    fn visit_export_decl<'ast: 'r, 'r>(
+        &mut self,
+        export_decl: &'ast ExportDecl,
+        ast_path: &mut AstNodePath<AstParentNodeRef<'r>>,
+    ) {
+        match &export_decl.decl {
+            Decl::Var(box VarDecl { kind, decls, .. }) if kind == &VarDeclKind::Const => {
+                for decl in decls {
+                    if let VarDeclarator {
+                        name: Pat::Ident(BindingIdent { id, .. }),
+                        init: Some(init),
+                        ..
+                    } = decl
+                    {
+                        let id = id.sym.as_ref();
+                        if let Some(prop) = self.properties.get_mut(id) {
+                            *prop = extract_value(&self.expr_ctx, init, id.to_string());
+                        };
+                    }
+                }
+            }
+            _ => {}
+        }
+
+        export_decl.visit_children_with_path(self, ast_path);
+    }
+}
+
+/// Coerece the actual value of the given ast node.
+fn extract_value(ctx: &ExprCtx, init: &Expr, id: String) -> Option<Const> {
+    match init {
+        init if init.is_undefined(ctx) => Some(Const::Value(Value::Null)),
+        Expr::Ident(ident) => Some(Const::Unsupported(format!(
+            "Unknown identifier \"{}\" at \"{}\".",
+            ident.sym, id
+        ))),
+        Expr::Lit(lit) => match lit {
+            Lit::Num(num) => Some(Const::Value(Value::Number(
+                Number::from_f64(num.value).expect("Should able to convert f64 to Number"),
+            ))),
+            Lit::Null(_) => Some(Const::Value(Value::Null)),
+            Lit::Str(s) => Some(Const::Value(Value::String(s.value.to_string()))),
+            Lit::Bool(b) => Some(Const::Value(Value::Bool(b.value))),
+            Lit::Regex(r) => Some(Const::Value(Value::String(format!(
+                "/{}/{}",
+                r.exp, r.flags
+            )))),
+            _ => Some(Const::Unsupported("Unsupported Literal".to_string())),
+        },
+        Expr::Array(arr) => {
+            let mut a = vec![];
+
+            for elem in &arr.elems {
+                match elem {
+                    Some(elem) => {
+                        if elem.spread.is_some() {
+                            return Some(Const::Unsupported(format!(
+                                "Unsupported spread operator in the Array Expression at \"{}\"",
+                                id
+                            )));
+                        }
+
+                        match extract_value(ctx, &elem.expr, id.clone()) {
+                            Some(Const::Value(value)) => a.push(value),
+                            _ => {
+                                return Some(Const::Unsupported(
+                                    "Unsupported value in the Array Expression".to_string(),
+                                ))
+                            }
```

Comment:
It should probably propagate the inner Unsupported error message.
```suggestion
                            Some(Const::Value(value)) => a.push(value),
                            Some(Const::Unsupported(message)) => {
                                return Some(Const::Unsupported(
                                    format!("Unsupported value in the Array Expression: {message}"),
                                ))                            
                            },
                            _ => {
                                return Some(Const::Unsupported(
                                    "Unsupported value in the Array Expression".to_string(),
                                ))
                            }
```